### DISCUSSION TITLE
Delete and Add files in GitHub Repository using Py Scripting and GH Actions schedule

### DISCUSSION THREAD

**Discussion by jairamjidgekar** [community/community#37954] [Actions] [answered]
Hi, 

I have a .py script, which deletes an existing file and replaces with the new file (in the same path). I have scheduled this using GH Actions on a daily basis. 

On any given day, there might be chances that, there won't be any changes in the file downloaded. So, in that situation the GH Actions workflow will fail in the commit task, since there's no change. 

------------------------------------------------------------------------------------------------------------
Error message: 
------------------------------------------------------------------------------------------------------------
```
Your branch is up to date with 'origin/master'.
nothing to commit, working tree clean
##[error]Process completed with exit code 1.
```
------------------------------------------------------------------------------------------------------------

As a work around, what I did is, I separated the code (delete files and add files) into two different .py scripts and added commit tasks respectively. 

------------------------------------------------------------------------------------------------------------
Code snippet:
------------------------------------------------------------------------------------------------------------

```
 - id: python_remove
        name: Python Remove file execution
        run: python Images/Remove_Files.py  
          # python Images/Images_of_ILO_Biography.py
      
      # Remove old files in folder,  push
      - id: remove_commit
        name: Commit files - Removes
        run:
          git config --local user.name actions-user

          git config --local user.email "actions@github.com" 
          
          git fetch

          git add *

          git commit -am "GH Action   File removed    $(date)" 
 
          git push -f origin main


 - id: python 
        name: Python file execution 
        run: python Images/Images_of_ILO_Biography.py  
          # python Images/Images_of_ILO_Biography.py
      
      # Add files in folder,  push
      - id: commit_files
        name: Commit files  
        run:
          git config --local user.name actions-user

          git config --local user.email "actions@github.com" 
          
          git fetch

          git add *

          git commit -am "GH Action   Files added    $(date)" 
 
          git push -f origin main
```

But this code seems to be recurring (I feel there should be some other option, if any). 

Can you please let me know how can we overcome this situation?

Thanks,
Jairam P.

**Comment by airtower-luna** [comment, marked as answer]
Instead of deleting/adding files, *check if there are changes* before making a commit. You can use [`git diff --exit-code`](https://git-scm.com/docs/git-diff#Documentation/git-diff.txt---exit-code):

```yaml
      - id: commit_files
        name: Commit files  
        run:
          git config --local user.name actions-user
          git config --local user.email "actions@github.com"
          git fetch
          if ! git diff --exit-code; then
              git add *
              git commit -am "GH Action   Files added    $(date)"
              git push -f origin main
          fi
```

I'd recommend against using `git add *` though, try to be specific about the files you intend to add (and then check only those with `git diff`).

**Reply by jairamjidgekar** [reply]
Hello, 

Thank you very much for the response. 

I would try to make the changes today and let you know how it goes. 

Regarding - ` git add * `, I have multiple .py script  (say 10-15 nos) executing in the GH Actions. 
Hence, I have used ` * `. 

Thanks again, 
Jairam P.

**Reply by jairamjidgekar** [reply]
This is indeed working! thank you. 

But if you see the workflow log, ` git diff --exit-code` displays all the differences in the wf log file. 

Instead of ` --exit-code `, I have used ` --quiet ` which does the same, but suppresses the differences.
